lib/tokenize.cpp

```diff
@@ -6225,6 +6225,28 @@ void Tokenizer::simplifyHeadersAndUnusedTemplates()
     for (Token *tok = list.front(); tok; tok = tok->next()) {
         const bool isIncluded = (tok->fileIndex() != 0);
 
+        // Remove constructor initializer
+        if (isIncluded && !mSettings.checkHeaders && Token::Match(tok, "%name% (")) {
+            Token *start = tok->next()->link();
+            if (start) {
+                start = start->next();
+            }
+            if (start && start->str() == ":") {
+                const Token *next = start;
+                while (Token::Match(next, "[,:] %name%")) {
+                    next = next->next(); // Go to %name%
+                    next = next->next(); // Go to opening '(' or '{'
```

Comment:
please consider if baseclass is declared in a different namespace also:

    C() : a::base(0) {}